circleci/resource_circleci_environment_variable_test.go

```diff
@@ -128,6 +136,165 @@ func TestCircleCIEnvironmentVariableCreateAlreadyExists(t *testing.T) {
+					"value",
+				},
+			},
+		},
+	})
+}
+
+func TestParseEnvironmentVariableId(t *testing.T) {
+	organization := acctest.RandString(8)
+	envName := acctest.RandString(8)
+	projectNames := []string{
+		"TEST_" + acctest.RandString(8),
+		"TEST-" + acctest.RandString(8),
+		"TEST." + acctest.RandString(8),
+		"TEST_" + acctest.RandString(8) + "." + acctest.RandString(8),
+		"TEST-" + acctest.RandString(8) + "." + acctest.RandString(8),
+		"TEST." + acctest.RandString(8) + "." + acctest.RandString(8),
+	}
+
+	for _, name := range projectNames {
+		expectedId := fmt.Sprintf("%s.%s.%s", organization, name, envName)
+		actualOrganization, actualProjectName, actualEnvName := parseEnvironmentVariableId(expectedId)
+		assert.Equal(t, organization, actualOrganization)
+		assert.Equal(t, name, actualProjectName)
+		assert.Equal(t, envName, actualEnvName)
+	}
+}
+
+func testCircleCIEnvironmentVariableResourceOrgStateDataV0(organization, project, name string) map[string]interface{} {
+	return map[string]interface{}{
+		"id":           name,
+		"name":         name,
+		"project":      project,
+		"organization": organization,
+	}
+}
+
+func testCircleCIEnvironmentVariableNoOrgStateDataProviderOrgV0(project, name string) map[string]interface{} {
+	return map[string]interface{}{
+		"id":      name,
+		"name":    name,
+		"project": project,
+	}
+}
+
+func testCircleCIEnvironmentVariableResourceOrgStateDataV1(organization, project, name string) map[string]interface{} {
+	v0 := testCircleCIEnvironmentVariableResourceOrgStateDataV0(organization, project, name)
+	return map[string]interface{}{
+		"id":           fmt.Sprintf("%s.%s.%s", v0["organization"].(string), v0["project"].(string), v0["name"].(string)),
+		"name":         v0["name"].(string),
+		"project":      v0["project"].(string),
+		"organization": v0["organization"].(string),
+	}
+}
+
+func testCircleCIEnvironmentVariableNoOrgStateDataProviderOrgV1(organization, project, name string) map[string]interface{} {
+	v0 := testCircleCIEnvironmentVariableNoOrgStateDataProviderOrgV0(project, name)
+	return map[string]interface{}{
+		"id":      fmt.Sprintf("%s.%s.%s", organization, v0["project"].(string), v0["name"].(string)),
+		"name":    v0["name"].(string),
+		"project": v0["project"].(string),
+	}
+}
+
+func TestCircleCIEnvironmentVariableResourceOrgStateUpgradeV0(t *testing.T) {
+	project := os.Getenv("CIRCLECI_PROJECT")
+	envName := "TEST_" + acctest.RandString(8)
+	organization := os.Getenv("TEST_CIRCLECI_ORGANIZATION")
+	providerClient, _ := NewConfig("foo", os.Getenv("CIRCLECI_VCS_TYPE"), "http://example.com")
+	actual, err := resourceCircleCIEnvironmentVariableUpgradeV0(testCircleCIEnvironmentVariableResourceOrgStateDataV0(organization, project, envName), providerClient)
+	if err != nil {
+		t.Fatalf("error migrating state: %s", err)
+	}
+
+	expected := testCircleCIEnvironmentVariableResourceOrgStateDataV1(organization, project, envName)
+	if !reflect.DeepEqual(expected, actual) {
```

Comment:
another nit: given that you already import testify, can you replace these `reflect.DeepEqual`s with `assert.Equal` for consistency? While you are at it use also `assert.NoError` for errors :P

Reply:
Whoops. I just wholesale copied this part from the TF docs